frontend/src/bundles/common/middlewares/error-handling.middleware.ts

```diff
@@ -0,0 +1,47 @@
+import { createStandaloneToast } from '@chakra-ui/react';
+import {
+    type Middleware,
+    isRejected,
+    isRejectedWithValue,
+} from '@reduxjs/toolkit';
+import { type ServerValidationErrorResponse } from 'shared';
+
+import { theme } from '~/framework/theme/theme.js';
+
+import { stringToReactNode } from '../helpers/helpers.js';
+
+const { toast } = createStandaloneToast({ theme: theme });
+const toastId = 'redux-store-error';
+
+const errorMiddleware: Middleware = () => {
+    return (next) => (action) => {
+        let message: string = '';
+        if (isRejectedWithValue(action)) {
+            message += JSON.stringify(action.payload);
+        } else if (isRejected(action)) {
+            const error = action.error as ServerValidationErrorResponse;
+            message += `${error.message}\n`;
+            if (error.details) {
+                for (const errorDetail of error.details) {
+                    message += `\t- ${errorDetail.message}\n`;
+                }
+            }
+        }
+
+        if (message !== '' && !toast.isActive(toastId)) {
+            toast({
+                id: toastId,
+                title: 'An error occurred.',
+                description: stringToReactNode(message),
+                status: 'error',
+                duration: 7000,
+                isClosable: true,
+                position: 'top-right',
+            });
```

Comment:
Create a separate service for Toast that we could call for multiple scenarios: error, warn, success, etc.

Reply:
### I implemented a toast service for success, warning, error, info, and loading messages.
![6](https://github.com/user-attachments/assets/042624f1-7910-4cb7-b378-c880980d83a6)